python/paddle/fluid/tests/unittests/test_put_along_axis_op.py

```diff
@@ -0,0 +1,127 @@
+#   Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+import unittest
+import numpy as np
+import copy
+from op_test import OpTest, skip_check_grad_ci
+import paddle
+import paddle.fluid as fluid
+from paddle.framework import core
+from paddle.fluid.dygraph.base import switch_to_static_graph
+
+paddle.enable_static()
+
+
+@skip_check_grad_ci("backward calculation has wrong logic in op test")
+class TestPutAlongAxisOp(OpTest):
+    def setUp(self):
+        self.init_data()
+        self._cpu_only = True
```

Comment:
Should we also check gpu? As I found your GPU kernel bug (see my comment above)

Reply:
Done.